plugins/notifications-backend/src/service/router.ts

```diff
@@ -1,62 +1,76 @@
 import { errorHandler } from '@backstage/backend-common';
+import { CatalogClient } from '@backstage/catalog-client';
+import { Config } from '@backstage/config';
 
 import express from 'express';
 import Router from 'express-promise-router';
-import { Knex } from 'knex';
 import { Logger } from 'winston';
 
-export interface RouterOptions {
+import { initDB } from './db';
+import {
+  createNotification,
+  getNotifications,
+  getNotificationsCount,
+} from './handlers';
+
+interface RouterOptions {
   logger: Logger;
-  dbClient?: Knex<any, any[]>;
+  dbConfig: Config;
+  catalogClient: CatalogClient;
 }
 
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, dbClient } = options;
+  const { logger, dbConfig, catalogClient } = options;
 
   const router = Router();
   router.use(express.json());
 
-  if (!dbClient) {
-    logger.error('Missing dbClient');
-    return router;
+  if (!dbConfig) {
+    logger.error('Missing dbConfig');
+    throw new Error('Missing database config');
   }
 
+  // create DB client and tables
+  const dbClient = await initDB(dbConfig);
+
+  // rest endpoints/operations
   router.get('/health', (_, response) => {
-    logger.info('PONG!');
     response.json({ status: 'ok' });
   });
 
-  router.get('/notifications/count', async (_, response) => {
-    let msgcount = -1;
-    await dbClient('messages')
-      .count('* as CNT')
-      .then(count => {
-        msgcount = (count[0]?.CNT as unknown as number) || -1;
-      });
-    response.json({ status: 'ok', count: msgcount });
+  router.get('/notifications/count', (request, response) => {
+    getNotificationsCount(dbClient, request.query).then(result =>
+      response.json(result),
+    );
   });
 
-  router.post('/notifications', async (request, response) => {
-    let msgid = -1;
-    await dbClient('messages')
-      .insert(request.body)
-      .returning('id')
-      .then(id => {
-        // We should harden the type
-        msgid = id as unknown as number;
-      });
-    response.json({ status: 'ok', msgid: msgid });
+  router.get('/notifications', (request, response) => {
+    const { pageSize, pageNumber } = request.query;
+
+    if (typeof pageSize !== 'string' || typeof pageNumber !== 'string') {
+      throw new Error(
+        'either pageSize or pageNumber query string parameters are missing/invalid',
+      );
+    }
+
+    const pageSizeNum = Number.parseInt(pageSize);
```

Comment:
To make the linter happy:

```suggestion
    const pageSizeNum = Number.parseInt(pageSize, 10);
```